pex/pex.py

```diff
@@ -155,9 +155,14 @@ def minimum_sys_modules(cls, site_libs, modules=None):
 
   @classmethod
   def minimum_sys_path(cls, site_libs, inherit_path):
+    ver = sys.version[:3]
     scrub_paths = OrderedSet()
     site_distributions = OrderedSet()
     user_site_distributions = OrderedSet()
+    undesirable_paths = OrderedSet([
+      '/System/Library/Frameworks/Python.framework/Versions/%s/Extras/lib/python' % ver,
```

Comment:
PEX.site_libs() first path `/Library/Python/2.7/site-package` contains `Extras.pth` and `six.py` which seems to be causing most of the headache.
```sh
jvenus-signal-mbp13:pex jvenus$ env PEX_INTERPRETER=1 PEX_IGNORE_RCFILES=1 PEX_PYTHON=/usr/bin/python2.7 pex --python=/usr/bin/python2.7 pex
Python 2.7.10 (default, Jul 30 2016, 18:31:42)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from pex.pex import PEX
>>> PEX.site_libs()
set(['/Library/Python/2.7/site-packages', '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/site-python', '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages'])
```

```sh
jvenus-signal-mbp13:pex jvenus$ ls /Library/Python/2.7/site-packages/
Extras.pth                       README                           six-1.9.0-py2.7.egg-info         six.pyc                          vboxapi-1.0-py2.7.egg-info
RBTools-0.7.2.dev-py2.7.egg-info rbtools                          six.py                           vboxapi
jvenus-signal-mbp13:pex jvenus$ cat /Library/Python/2.7/site-packages/Extras.pth
/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python
/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/PyObjC
```